### DISCUSSION TITLE
How to embed terminal/vscode

### DISCUSSION THREAD

**Discussion by sebbycorp** [colinwilson/lotusdocs#146] [General]
I want to embed a terminal or vscode into a docs page.

I will have a docker of vscode server running and i want to import the gui into the docs section ? any thoughts on this?

**Comment by sebbycorp** [comment]
I guess i really just want to add a frame to docs and insert vscode browser 

<div class="code-server-frame">
    <iframe src="URL_OF_YOUR_VS_CODE_SERVER" width="100%" height="500px"></iframe>
</div>

where can i put this ?

**Comment by sn00py1310** [comment]
You could use a simple shortcode.
Something like this in `layouts/shortcodes/iframe.html`
```html
<iframe src="{{ .Get 0 }}"></iframe>
```

And use it like this
```md
---
title: "Your title"
description: "Your description, you can add more fields below of course..."
---
Some text content.

{{< iframe https://example.com/ >}}
```

For further information see:
- https://stackoverflow.com/a/71370488
- https://gohugo.io/templates/shortcode-templates/
